### ISSUE TITLE
Duplicated condition warning when its not duplicated

### ISSUE THREAD

**Issue by dpauli** [kalessil/phpinspectionsea#306]
Hi guys,

I found a little issue today in my the inspector.
```php
        $a = true;
        $b = true;

        if ($a === false && $b === false) {
            echo 'ab';
        } elseif ($a === false) {
            echo 'a';
        } elseif ($b === false) {
            echo 'b';
        }
```

This code is not useful (because only for test purposes). It shows 'This condition is duplicated in another if/elseif branch.` in each `elseif` branch. This is not right here.

**Comment by rentalhost** [comment]
It'll depends of context. In this context, really make sense do not warn about that, once that we are checking for static types (eg. `=== false`). But for calls, it make sense. For instance:

```php
if (check1() && check2()) {
}
else if (check1()) {
}
```

In this example, you should store the expression in a variable then check it to avoid double running it (except if it is intentional, but in this case, you should make it clear).

Backing to your context, I think about two options:

* You fix your code by storing the value on a variable (that should resolves the major part of this issue);
* We could consider create a sub-option on this inspection to not warn on comparison with static types;

**Comment by dpauli** [comment]
I see the benefit in not calling the function twice. In my case I have static variables and their should not be this warning.

Its not essential important to fix this, I only want to add this information here, so it can be fixed in future to make this tool better. I love this plugin :)

**Comment by rentalhost** [comment]
@kalessil How you should fix that? Allowing comparison with static types, maybe?

**Comment by kalessil** [comment]
Working on this, the idea is to allow variable comparison to bool

**Comment by rentalhost** [comment]
And for another scalar types? For instance: `$count === 1`.

**Comment by kalessil** [comment]
Out of the scope ATM
